### PR TITLE
Separate message delivery info for destination type

### PR DESCRIPTION
**Pull request by akalankapagoda**
Move MessageDeliveryInfo class to a separate file instead of an inner class.
Then forward MessageDeliveryInfo objects from SlotDeliveryWorker to MessageFlusher to delivery messages instead of sending parameters separately and selecting the MessageDeliveryInfo class from within MessageFlusher.

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageDeliveryInfo.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (sajinidesilva) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageDeliveryInfo.java]: Copy right year should be 2016

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageFlusher.java [hunk 2]
```diff
@@ -422,97 +282,83 @@ public void addAlreadyTrackedMessagesToBuffer(String destination, ProtocolType p
...
-            } catch (InterruptedException e1) {
-                //silently ignore
+                if (log.isDebugEnabled()) {
+                    log.debug("Sending messages from buffer num of msg = "
+                            + messageDeliveryInfo.getSizeOfMessageBuffer());
+                }
+                sendMessagesToSubscriptions(messageDeliveryInfo);
+                sentFromBuffer = true;
+
+            } catch (Exception e) {
```
Comment (sajinidesilva) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageFlusher.java]: Why are we catching an Exception here?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageFlusher.java [hunk 3]
```diff
@@ -422,97 +282,83 @@ public void addAlreadyTrackedMessagesToBuffer(String destination, ProtocolType p
...
+                /**
+                 * When there is a error, we will wait to avoid looping.
+                 */
+                long waitTime = queueWorkerWaitInterval;
+                failureCount++;
+                long faultWaitTime = Math.max(waitTime * 5, failureCount * waitTime);
+                try {
+                    Thread.sleep(faultWaitTime);
+                } catch (InterruptedException e1) {
```
Comment (sajinidesilva) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageFlusher.java]: rename e1 to ignore

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageFlusher.java (hunk 2 above)
Comment (akalankapagoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageFlusher.java]: This is legacy code modified by surrounding it by an if. Will remove this.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageFlusher.java (hunk 3 above)
Comment (akalankapagoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageFlusher.java]: Yes, will do.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageDeliveryInfo.java (hunk 1 above)
Comment (akalankapagoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageDeliveryInfo.java]: Updated.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageDeliveryInfo.java [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+     */
+    private String destination;
+
+    /**
+     * Subscription iterator
+     */
+    private Iterator<LocalSubscription> iterator;
+
+    /**
+     * in-memory message list scheduled to be delivered. These messages will be flushed
```
Comment (sajinidesilva) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageDeliveryInfo.java]: Shall we start comments with capital letters?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/CommonDeliveryRule.java [hunk 5]
```diff
@@ -30,8 +30,11 @@
      * necessary actions should be taken
      *
      * @param message message to evaluate rules
+     * @param protocolType The protocol type of the message
```
Comment (sajinidesilva) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/CommonDeliveryRule.java]: Shall we start comments with capital letters?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageFlusher.java [hunk 6]
```diff
@@ -422,97 +283,67 @@ public void addAlreadyTrackedMessagesToBuffer(String destination, ProtocolType p
...
-            long waitTime = queueWorkerWaitInterval;
-            failureCount++;
-            long faultWaitTime = Math.max(waitTime * 5, failureCount * waitTime);
-            try {
-                Thread.sleep(faultWaitTime);
-            } catch (InterruptedException e1) {
-                //silently ignore
+            if (log.isDebugEnabled()) {
+                log.debug("Sending messages in buffer destination= " + messageDeliveryInfo.getDestination());
+            }
```
Comment (sajinidesilva) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageFlusher.java]: Can;t we merge these two debug statements in to one?

Comment (akalankapagoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageFlusher.java]: yes, will update.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageDeliveryInfo.java (hunk 4 above)
Comment (akalankapagoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageDeliveryInfo.java]: Yes, will update.

**Pull request merged**